### DISCUSSION TITLE
<Leader>wm and wd does not trigger if type fast

### DISCUSSION THREAD

**Discussion by tranbinh1604** [LazyVim/LazyVim#4157] [Q&A]
Hello, 

I dont know just me or someone else having same situation. When I use `<leader>wd` to Delete window, or `<leader>wm` to toggle maximize, if I type with no time delay all those keys (`leader` > `w` > `d`), then the key binding does not trigger/ execute, but if I type slower (at least wait for which-key pane show, `leader` > 1s > `w` > `d` or `m` then all work)

I tried to add keymaps to overwrite `<leader>wm` & `<leader>wd` but still same.

You can see the screencast for more detail. My configuration also quite simple and nothing related to `<leader>w` or `<C-w>`, another strange things is other keymaps in same groups `<leader>w` still work without delay needed (ex: `leader` > `w` > `s` will split the window ... )

![CleanShot 2024-07-24 at 08 34 26](https://github.com/user-attachments/assets/f1ddd83f-c728-405f-a12f-2a1bd70eeff2)

**Comment by folke** [comment]
Fixed that already. will make a new release

**Reply by iniw** [reply]
The issue still persists for me on the latest master.

**Comment by nipuna-perera** [comment]
Seeing the same on many of the <leader>w... keys

**Reply by folke** [reply]
can't reproduce. You must be doing something in your config that messes this up

**Comment by iniw** [comment]
Could be a terminal emulator thing? I'm using wezterm (without unix sockets) and ttimeout is set to false. Happens on both my Linux and MacOS computers.

**Reply by iniw** [reply]
Tho it did happen on alacritty too.

**Reply by Eckii24** [reply]
Also happens for me, using Wezterm under MacOS.
Getting errors for <Leader>wd from time to time (like shown in the screencast above), but never for <Leader>wc (what should do the same under the hood)

**Comment by iniw** [comment]
I doubt this is a terminal emulator issue - it still happens now that I'm using ghostty on both MacOS and Linux.

**Comment by cyaconi** [comment]
I am in the same situation, using alacritty, iterm, and ghostty. I've tried everything with no luck, and the issue only occurs with the `<leader>-w` combination (In my case, I use `<leader>-ww` to save, and the issue is quite frustrating.)

**Comment by iniw** [comment]
I thought it may have something to do with the shell, but I recently switched to nushell from zsh and continue to see this problem.

**Comment by iniw** [comment]
I suspect it may have something to do with how the keymaps are defined:

https://github.com/LazyVim/LazyVim/blob/7fb320700ad0162be33ce816f2ba6f9dda41883f/lua/lazyvim/config/keymaps.lua#L184-L190

The combination `<leader>wd` to delete a window is ambiguous, it could execute either `<C-W>c`, defined on line 188, which is the one we want, or it could fall through the general "Windows" case, defined on line 185, which wil execute `<C-W>d`. That key is actually `Show diagnostics under the cursor`, hence the `E388: Couldn't find definition` error that we see being thrown in the video.

This may be the case for other keymaps as well.

It may have something to with the "timeout" family of options `timeout`, `timeoutlen` and `ttimeout`, which in my config are `true`, `300` and `false`, respectively. Though I'm pretty sure `which-key`'s timeout is independent from nvim.

@folke thoughts?
